messagingpush/src/main/java/io/customer/messagingpush/provider/FCMTokenProviderImpl.kt

```diff
@@ -0,0 +1,58 @@
+package io.customer.messagingpush.provider
+
+import android.content.Context
+import com.google.android.gms.common.ConnectionResult
+import com.google.android.gms.common.GoogleApiAvailability
+import com.google.firebase.messaging.FirebaseMessaging
+import io.customer.sdk.device.DeviceTokenProvider
+import io.customer.sdk.util.Logger
+
+/**
+ * Wrapper around FCM SDK to make the code base more testable. There is no concept of checked-exceptions in Kotlin
+ * so we need to handle the exception manually.
+ */
+class FCMTokenProviderImpl(
+    private val logger: Logger,
+    private val context: Context
+) : DeviceTokenProvider {
+
+    override fun isValidForThisDevice(context: Context): Boolean {
+        return try {
+            (
+                GoogleApiAvailability.getInstance()
+                    .isGooglePlayServicesAvailable(context) == ConnectionResult.SUCCESS
+                ).also {
+                logger.info("Is Firebase available on on this device -> $it")
+            }
+        } catch (exception: Throwable) {
+            logger.error(exception.message ?: "error checking google play services availability")
+            false
+        }
+    }
```

Comment:
`try/catch` would just give you an error, this method would probably narrow it down for you. 

From the google docs,

>Apps that rely on the Play Services SDK should always check the device for a compatible Google Play services APK before accessing Google Play services features

Regarding the 
> Why does the SDK have google play services installed

It shouldn't, think it was added to `Initialize` google services/firebase for the app who didn't before, but it doesn't. So i'll remove it, i thought i did already so good catch.

Reply:
If we were not fetching the token from the firebase, I would have agreed but since we do fetch it and we do mention it in docs as well, I think the trade-off isn't bad either. 

Getting push messages reliably is one of the core component, and secondly we just using `com.google.android.gms:play-services-base` which is the lightest of the depency, and is only going to be applicable if customer add the `push-messaging-fcm` module.

Reply:
Generally, I would agree to avoid optional dependency, but since the base library has no additional dependency and (size is `less than 400 bytes` ) and provides the vital information of `play services` being available, which is the core of fcm, I don't see any harm in having it. 

In my past experience, and even know, whenever I am told, my device token was not sent, my first question is, is google services available on your device, same happened with Rehan onboarding. 

So, my strong vote is to have it, specially due to size, but if you feel very strongly against it as well. I can ask the squad and have their take on it.

Reply:
It looks like [the play services base](https://mvnrepository.com/artifact/com.google.android.gms/play-services-base/18.0.1) library has 5 dependencies inside of it? Are you seeing something else saying that this library doesn't have any additional dependencies inside of it? 

I would like to hear what the squad thinks about this idea. I would like to hear how they feel about our code debugging customer's apps and using another dependency.

Reply:
Discussion : https://customerio.slack.com/archives/C01QYDKD0SU/p1657095131332079

Reply:
@levibostian this PR is still blocked, mind reviewing it again, after looking into the discussion thread?